### PR TITLE
Add auth support

### PR DESCRIPTION
**Pull request by damccorm**

### CODE REVIEW

**Review by stephenmichaelf — APPROVED** [review]

**Comment by ClintyB** [comment]
auth[](url)

File: lib/authutil.js [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    if (scope && scope[0] != '@') {
+        scope = '@' + scope;
+    }
+    core.debug(`Setting auth in ${fileLocation}`);
+    let newContents = '';
+    if (fs.existsSync(fileLocation)) {
+        const curContents = fs.readFileSync(fileLocation, 'utf8');
+        curContents.split(os.EOL).forEach((line) => {
+            // Add current contents unless they are setting the registry
+            if (!line.toLowerCase().startsWith('registry')) {
```
Comment (joebowbeer) [inline lib/authutil.js]: This doesn't seem right. Only the line(s) that is being redefined should be removed.

`registry` defines the default registry for unscoped dependencies, whereas if scope is provided this function is defining a scoped registry.

Comment (stephenmichaelf) [reply lib/authutil.js]: FYI @hross

**Review by Sk21renee37 — APPROVED** [review]

**Comment by aguilarj0987-oss** [comment]
yes>

**Review by pelonducks25-crypto — APPROVED** [review]

**Pull request merged**